internal/util/cryptsetup.go

```diff
@@ -18,22 +18,41 @@ package util
 
 import (
 	"bytes"
+	"context"
+	"errors"
 	"fmt"
 	"os"
 	"os/exec"
 	"strconv"
 	"strings"
+	"time"
 
 	"github.com/ceph/ceph-csi/internal/util/file"
 	"github.com/ceph/ceph-csi/internal/util/log"
 )
 
-// Limit memory used by Argon2i PBKDF to 32 MiB.
-const cryptsetupPBKDFMemoryLimit = 32 << 10 // 32768 KiB
+const (
+	// Maximum time to wait for cryptsetup commands to complete.
+	CryptSetupExecutionTimeout = 2*time.Minute + 30*time.Second
+
+	// Limit memory used by Argon2i PBKDF to 32 MiB.
+	cryptsetupPBKDFMemoryLimit = 32 << 10 // 32768 KiB
+)
+
+// LuksWrapper is a struct that provides a context-aware wrapper around cryptsetup commands.
+type LuksWrapper struct {
+	ctx context.Context
+}
+
+// NewLuksWrapperWithContext creates a new LuksWrapper instance with the provided context.
+// The context is used to control the lifetime of the cryptsetup commands executed by the LuksWrapper.
+func NewLuksWrapperWithContext(ctx context.Context) *LuksWrapper {
+	return &LuksWrapper{ctx: ctx}
+}
 
 // LuksFormat sets up volume as an encrypted LUKS partition.
-func LuksFormat(devicePath, passphrase string) (string, string, error) {
-	return execCryptsetupCommand(
+func (l *LuksWrapper) LuksFormat(devicePath, passphrase string) (string, string, error) {
```

Comment:
`LuksFormat` and other functions are now part of a LUKS type, no need to include `Luks` in every function name. Instead, rename the functions like `func (l *luksWrapper) Format(...`